theia-turn/turn-sprotty/src/views/ucm/connection.edge.tsx

```diff
@@ -53,4 +170,37 @@ export class ConnectionEdgeView implements IView {
     protected renderDanglingEdge(message: string, edge: SEdge, context: RenderingContext): VNode {
         return <text class-sprotty-edge-dangling={true}>{message}</text>;
     }
+
+    protected isGoingUp(firstPoint: Point, secondPoint: Point): boolean {
+        return firstPoint.y > secondPoint.y;
+    }
+
+    protected isGoingRight(firstPoint: Point, secondPoint: Point): boolean {
+        return secondPoint.x > firstPoint.x;
+    }
+
+    protected getDownPath(arcRadius: number, p: Point): String {
+        return this.getLinePath(p.x, p.y - arcRadius);
+    }
+
+    protected getUpPath(arcRadius: number, p: Point): String {
+        return this.getLinePath(p.x, p.y + arcRadius);
+    }
+
+    protected getLeftPath(arcRadius: number, p: Point): String {
+        return this.getLinePath(p.x + arcRadius, p.y);
+    }
+
+    protected getRightPath(arcRadius: number, p: Point): String {
+        return this.getLinePath(p.x - arcRadius, p.y);
+    }
```

Comment:
Hey @Gabetn , creating specific functions for the different getArcPathes would involve many functions since an arc path needs to consider the previous direction to decide which way to arc (UpRight, LeftDown, LeftUp, ... ). If we have time we could create these functions.